core/src/components/banner/banner.tsx

```diff
@@ -114,14 +118,15 @@ export class TdsBanner {
             <tds-icon name={this.icon} size="20px"></tds-icon>
           </div>
         )}
-        <div class={`banner-content ${this.type} ${!this.icon ? 'no-icon' : ''}`}>
-          {this.header && <span class={`banner-header`}>{this.header}</span>}
-          {this.hasSubheader && <slot name="banner-subheader"></slot>}
-          {this.hasLink && (
-            <div class={`banner-link ${!this.hasSubheader ? 'no-subheader' : ''}`}>
-              <slot name="banner-link"></slot>
-            </div>
-          )}
+
+        <div class="content">
+          <div class="header-subheader">
+            {this.header && <div class="header">{this.header}</div>}
+            {usesHeaderSlot && <slot name="header"></slot>}
```

Comment:
Yeah let's talk it through at a checkin. This would only go for the header though. Since we can't render anything for the subheader if it isn't chosen, cause that would make the gap persist even if there is no subheader.

Reply:
So I guess the question is less code or consistency?

Reply:
Lets go with current solution and inform the team about discussion we had in a call ;)